silk/tests/openthread/ot_test_meshcop_joiner_discerner.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import unittest
+
+import silk.hw.hw_resource as hwr
+import silk.node.fifteen_four_dev_board as ffdb
+import silk.tests.testcase as testcase
+from silk.config import wpan_constants as wpan
+from silk.tools import wpan_table_parser
+from silk.tools.wpan_util import verify, verify_within, is_associated
+from silk.utils import process_cleanup
+
+hwr.global_instance()
+WAIT_TIME = 2  # seconds
+
+PSK1 = 'UNCHARTEDTHEL0STLEGACY'
+PSK2 = 'UNCHARTED4ATH1EFSEND'
+PSK3 = 'THELAST0FUS'
```

Comment:
Maybe these should be random strings?

Reply:
Ack.

Reply:
Tried random keys but got error. 

e.g. PSK2=random_string(20)
[2020-10-30 23:04:29,618] [silk.Dev-ABDD] [DEBUG] sudo ip netns exec ttyACM10 /usr/local/bin/wpanctl -I ttyACM10 joiner --join FXHHBGNZSMPLRWXRJFDC
[2020-10-30 23:04:29,641] [silk.Dev-ABDD] [DEBUG] [stdout] PSKd "FXHHBGNZSMPLRWXRJFDC" must not contain I, O, Q, or Z characters
[2020-10-30 23:04:29,642] [silk.Dev-ABDD] [DEBUG] [stdout] joiner: error: Invalid PSKd 2

Thus let's keep them as such.